### PR TITLE
refactor complex functions

### PR DESCRIPTION
**Pull request by tobiaskleiner**

### CODE REVIEW

File: simtools/applications/db_add_file_to_db.py [hunk 1]
```diff
@@ -50,47 +50,46 @@
 from simtools.db import db_handler
 
 
-def main():
+def initialize_config():
```
[the remark this answers is not in the document]
Comment (tobiaskleiner) [reply simtools/applications/db_add_file_to_db.py]: Changed to _parse.

File: simtools/applications/db_add_file_to_db.py [hunk 2]
```diff
@@ -50,47 +50,46 @@
...
         nargs="+",
     )
     group.add_argument(
         "--input_path",
-        help=(
-            "A directory with files to upload to the DB. "
-            "All files in the directory with the following extensions "
-            f"will be uploaded: {', '.join(_db_tmp.ALLOWED_FILE_EXTENSIONS)}"
-        ),
+        help=("A directory with files to upload to the DB. ..."),
```
[the remark this answers is not in the document]
Comment (tobiaskleiner) [reply simtools/applications/db_add_file_to_db.py]: Removed

File: simtools/applications/db_add_file_to_db.py [hunk 3]
```diff
@@ -104,22 +103,32 @@ def main():
...
         raise ValueError("No files were provided to upload")
-    if len(files_to_insert) == 1:
-        plural = ""
-    else:
-        plural = "s"
+
+    return files_to_insert
+
+
+def confirm_and_insert_files(files_to_insert, args_dict, db, logger):
```
[the remark this answers is not in the document]
Comment (tobiaskleiner) [reply simtools/applications/db_add_file_to_db.py:126]: Added docstring

File: simtools/applications/db_development_tools/mark_non_optics_parameters_non_applicable.py [hunk 4]
```diff
@@ -21,7 +21,7 @@
 from simtools.db import db_handler
 
 
-def main():
+def initialize_config():
```
[the remark this answers is not in the document]
Comment (tobiaskleiner) [reply simtools/applications/db_development_tools/mark_non_optics_parameters_non_applicable.py]: done

File: simtools/applications/derive_psf_parameters.py [hunk 5]
```diff
@@ -237,103 +224,160 @@ def add_parameters(
...
-        _, rmsd = run_pars(pars, plot=args_dict["plot_all"])
+        _, rmsd = run_pars(tel_model, args_dict, pars, data_to_plot, radius, pdf_pages)
         if rmsd < min_rmsd:
             min_rmsd = rmsd
             best_pars = pars
+
+    return best_pars, min_rmsd
+
+
+# pylint: disable=too-many-statements
```
[the remark this answers is not in the document]
Comment (tobiaskleiner) [reply simtools/applications/derive_psf_parameters.py]: Removed

**Comment by tobiaskleiner** [comment]
@GernotMaier thanks for the review. I have made the requested changes. Let me know if I can merge.

**Comment by GernotMaier** [comment]
Looks good - please merge. Thanks for this effort!

**Pull request merged**